air/src/execution_step/execution_context/stream_maps_variables.rs

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+pub(crate) mod errors;
+
+use crate::execution_step::ExecutionResult;
+use crate::JValue;
+
+use crate::execution_step::boxed_value::StreamMap;
+use crate::execution_step::Generation;
+use crate::execution_step::ValueAggregate;
+
+use air_interpreter_data::GenerationIdx;
+use air_interpreter_data::RestrictedStreamGens;
+use air_interpreter_data::RestrictedStreamMapGens;
+use air_parser::ast::Span;
+use air_parser::AirPos;
+use air_trace_handler::merger::ValueSource;
+use air_trace_handler::TraceHandler;
+use serde::Serialize;
+
+use std::collections::hash_map::Entry::{Occupied, Vacant};
+use std::collections::HashMap;
+
+use std::fmt;
+
```

Comment:
Let's add `TODO` here that this code should be simplified and unified with `stream_variables`